src/main/java/pl/uj/io/cuteanimals/model/action/InvestigateAction.java

```diff
@@ -26,7 +34,15 @@ public void setArgs(List<String> unused) {
 
     @Override
     public IResult execute(ICharacter character) {
-        // TODO: fetch current player location and return its "look around" method
-        return new Result("Looking around null");
+        if (!getAcceptableStates().contains(character.getCurrentGameState())) {
+            return new Result("This action cannot be executed now");
```

Comment:
Maybe change to sth like this:
```suggestion
            return new Result("This isn't time for this.");
```
Just to keep immersion.